diff --git a/node_modules/waku/dist/lib/utils/managed.js b/node_modules/waku/dist/lib/utils/managed.js
index a43ada1..866f50e 100644
--- a/node_modules/waku/dist/lib/utils/managed.js
+++ b/node_modules/waku/dist/lib/utils/managed.js
@@ -22,14 +22,16 @@ export default adapter(
 };
 export const getManagedClientEntry = ()=>{
     return `
-import { StrictMode, createElement } from 'react';
+import { StrictMode, createElement, startTransition } from 'react';
 import { createRoot, hydrateRoot } from 'react-dom/client';
 import { Router } from 'waku/router/client';
 
 const rootElement = createElement(StrictMode, null, createElement(Router));
 
 if (globalThis.__WAKU_HYDRATE__) {
-  hydrateRoot(document, rootElement);
+  startTransition(() => {
+    hydrateRoot(document, rootElement);
+  });
 } else {
   createRoot(document).render(rootElement);
 }
